manifests/server/config_entry.pp

```diff
@@ -98,10 +98,13 @@
   # is managed for us in postgresql::server::config.
   if $facts['os']['name'] == 'Debian' or $facts['os']['name'] == 'Ubuntu' {
     if $name == 'data_directory' {
+      $stop_command = ['service', $postgresql::server::service_name, 'stop']
+      $stop_onlyif = ['service', $postgresql::server::service_name, 'status']
+      $stop_unless = [['grep', "data_directory = '${shell_escape($value)}'", $postgresql::server::postgresql_conf_path]]
```

Comment:
I still think `shell_escape` should never be used when you pass it as an array. It's doing something twice: shell_escape is when you pass something through a shell, but passing it as an array doesn't spawn a shell at all.

Reply:
Have removed the shell_escapes where the resultant string is being passed as an array.
Have also clarified the ones one full string is being passed by removing the array braces around them